DD_TRACE_RETAIN_THREAD_CAPABILITIES option for PHP tracer (#11145)
DataDog/documentation: content/en/tracing/setup_overview/setup/php.md

: **Default**: `null`<br>
CSV of URI mappings to normalize resource naming for outgoing requests (see [Map resource names to normalized URI](#map-resource-names-to-normalized-uri)).

`DD_TRACE_RETAIN_THREAD_CAPABILITIES`
: **Default**: `false`<br>

> Only effective under linux. Set to `true` to retain capabilities on datadog background threads upon change of the effective user id. This option will not affect most setups, but some modules - to date Datadog is only aware of [apache's mod-ruid2](https://github.com/mind04/mod-ruid2) - may invoke setuid() or similar syscalls, leading to crashes or loss of functionality as it loses capabilities.<br>

Reviewer: ```suggestion
Works for Linux. Set to `true` to retain capabilities on Datadog background threads when you change the effective user ID. This option does not affect most setups, but some modules - to date Datadog is only aware of [Apache's mod-ruid2](https://github.com/mind04/mod-ruid2) - may invoke `setuid()` or similar syscalls, leading to crashes or loss of functionality as it loses capabilities.

```

> **Note:** Enabling this option may compromise security. This option, standalone, does not pose a security risk. However, an attacker being able to exploit a vulnerability in PHP or web server may be able to escalate privileges with relative ease, if the web server or PHP were started with full capabilities, as the background threads will retain their original capabilities. It is recommended to restrict the capabilities of the web server with the `setcap` utility.

Reviewer: ```suggestion
**Note:** Enabling this option may compromise security. This option, standalone, does not pose a security risk. However, an attacker being able to exploit a vulnerability in PHP or web server may be able to escalate privileges with relative ease, if the web server or PHP were started with full capabilities, as the background threads will retain their original capabilities. Datadog recommends restricting the capabilities of the web server with the `setcap` utility.
```
Should we document something that is a security risk? If we want to keep it, we should add much more details to this last sentence about how you use setcap and exactly what you set to lower your risk.

Author: What exactly you set depends on your needs. We don't know the customers setup and cannot decide that for him.
On whether to document it: It is at the very least some text our support team can link the customer. And maybe, if the customer is reading the config section carefully and using mod_ruid2, he can be immediately aware of this without any frustration on either side.
As the text states, it's not a direct risk, just a lowered layer of defense, once the application has been breached anyway.
